ReactWindows/ReactNative.Shared/UIManager/BorderedCanvasManager.cs

```diff
@@ -12,16 +13,58 @@ namespace ReactNative.UIManager
     /// View parent manager for bordered canvases.
     /// </summary>
     public abstract class BorderedCanvasManager<TCanvas> : BorderedViewParentManager<TCanvas>
-        where TCanvas : Canvas
+        where TCanvas : BorderedCanvas
     {
         /// <summary>
+        /// Checks if the Canvas has a Border already.
+        /// </summary>
+        protected override bool HasBorder(TCanvas view)
+        {
+            return view.Border != null;
+        }
+
+        /// <summary>
+        /// Adds a Border to a Canvas if it hasn't been added already.
+        /// </summary>
+        protected override Border GetOrCreateBorder(TCanvas view)
+        {
+            if (view.Border == null)
+            {
+                view.Border = new Border { BorderBrush = s_defaultBorderBrush };
+            }
+
+            return view.Border;
+        }
+
+        /// <summary>
+        /// Keeps the Border's dimensions in sync with with parent Canvas.
+        /// </summary>
+        public override void SetDimensions(TCanvas view, Dimensions dimensions)
+        {
+            base.SetDimensions(view, dimensions);
+
+            var border = view.Border;
+
+            if (border != null)
+            {
+                border.Width = dimensions.Width;
+                border.Height = dimensions.Height;
+            }
```

Comment:
Just curious, do you think this is a better approach? XAML binding also worked, I wasn't sure what felt like more of a hack. Do you think there's any performance benefits to either?